kayenta/kayenta-integration-tests/src/test/java/com/netflix/kayenta/configuration/EmbeddedPrometheusBootstrapConfiguration.java

```diff
@@ -13,66 +13,135 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package com.netflix.kayenta.configuration;
 
 import static com.playtika.test.common.utils.ContainerUtils.containerLogsConsumer;
 
-import com.netflix.kayenta.utils.EnvironmentUtils;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
 import java.time.Duration;
 import java.util.LinkedHashMap;
 import java.util.Map;
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.ApplicationListener;
 import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
 import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.Environment;
 import org.testcontainers.Testcontainers;
 import org.testcontainers.containers.GenericContainer;
 import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
 import org.testcontainers.containers.wait.strategy.WaitStrategy;
 import org.testcontainers.utility.MountableFile;
 
 @Slf4j
-@Configuration
+@TestConfiguration
+@RequiredArgsConstructor
 public class EmbeddedPrometheusBootstrapConfiguration {
 
-  // Exposes host machine port to be used by prometheus container for scraping metrics from
-  // /prometehus endpoint
-  // See for more details:
-  // https://www.testcontainers.org/features/networking/#exposing-host-ports-to-the-container
-  static {
-    Testcontainers.exposeHostPorts(8081);
+  private static final int PROMETHEUS_INTERNAL_PORT = 9090;
+  private final Environment environment;
+
+  @Bean
+  public WaitStrategy prometheusWaitStrategy() {
+    return new HttpWaitStrategy()
+        .forPath("/status")
+        .forPort(PROMETHEUS_INTERNAL_PORT)
+        .forStatusCode(200);
   }
 
-  private static final int PORT = 9090;
+  private GenericContainer<?> prometheusContainer;
```

Comment:
At the moment it looks like this is only used in the prometheus method, so it could be a local variable there...Maybe that changes once there's logic to stop the container when the application context goes away though.

Reply:
yes used this property to stop the container 
https://github.com/spinnaker/spinnaker/pull/7053/commits/935664dec54f435296c39f8dcd350f4110821063